cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -104,28 +125,33 @@ func mutateEtcdOrDie(mutator func() error) {
 	}
 }
 
-func newEtcdClient() (client *etcd.Client) {
-	maxConnectRetries := 12
-	for maxConnectRetries > 0 {
-		if _, err := tools.GetEtcdVersion(*etcd_server); err != nil {
-			log.Fatalf("Failed to connect to etcd server: %v, error: %v", *etcd_server, err)
-			if maxConnectRetries > 0 {
-				log.Println("Retrying request after 5 second sleep.")
-				time.Sleep(5 * time.Second)
-				maxConnectRetries--
-			} else {
-				return nil
-			}
-		} else {
-			log.Printf("Etcd server found: %v", *etcd_server)
+func newEtcdClient(etcdServer string) (*etcd.Client, error) {
+	var (
+		client *etcd.Client
+		err    error
+	)
+	retries := maxConnectRetries
+	for retries > 0 {
+		if _, err = tools.GetEtcdVersion(etcdServer); err == nil {
+			break
+		}
+		if maxConnectRetries == 1 {
```

Comment:
Not the end of the world... If maxConnectRetries == 1, it'll just never print out the log message, and when it runs out of retries, it'll wait an extra 5 seconds. @vishh Worth fixing in a followup, though.